packages/flutter_tools/lib/src/protocol_discovery.dart

```diff
@@ -95,8 +95,6 @@ class ProtocolDiscovery {
   void _located(Uri uri) {
     assert(_completer != null);
     assert(!_completer.isCompleted);
```

Comment:
It's not clear what we want to do here. From digging through the git history, it looks like the create-new-instance-on-complete logic has been around for a while. I suspect it was from before we did a good job of just getting logs from the current session, and we could have multiple matches to service protocol urls. Asserting's probably not the right thing to do now - perhaps just complete the completer if `! isCompleted`, and log if it is?